### PR TITLE
Add method to download artifacts

### PR DESCRIPTION
**Pull request by ostefano**
This PR adds methods to:
- download an artifact
- list the artifacts available (process snapshots, etc)

### CODE REVIEW

File: src/tau_clients/nsx_defender/__init__.py [hunk 1]
```diff
@@ -626,17 +668,128 @@ def get_completed(
...
+            [
+                {
+                    "task_uuid": "aba8dccf1641001000c9dd8526ac7c9d",
+                    "report_uuid": "4ffa644851ebe118wu1pbwlt3C44cwAYRtfdXY5L3gRhHJ_QANAyuQ",
+                    "artifact_name": "process_snapshots_2",
+                },
+                ...
+            ]
+        """
+        return self._get_result_artifact_names(
```
Comment (Phat3) [inline src/tau_clients/nsx_defender/__init__.py:711]: The function `_get_result_artifact_names` is not defined in the class. Did you mean `get_result_artifact_names`?

File: src/tau_clients/nsx_defender/__init__.py [hunk 2]
```diff
@@ -626,17 +668,128 @@ def get_completed(
...
+                    "artifact_name": "process_snapshots_2",
+                },
+                ...
+            ]
+        """
+        ret = []
+        if not report_types:
+            report_types = set([])
+        if not metadata_types:
+            metadata_types = set([])
```
Comment (Phat3) [inline src/tau_clients/nsx_defender/__init__.py:742]: In the parameter type annotation, these two variables are supposed to be lists but here they are created as sets. Which one is the correct type?

Comment (ostefano) [reply src/tau_clients/nsx_defender/__init__.py:742]: Replaced with iterable

File: src/tau_clients/nsx_defender/__init__.py (hunk 1 above)
Comment (ostefano) [reply src/tau_clients/nsx_defender/__init__.py:711]: Fixed

**Pull request closed without merging**